packages/block-library/src/navigation-menu/edit.js

```diff
@@ -45,6 +48,19 @@ function NavigationMenu( {
 	setTextColor,
 	setAttributes,
 } ) {
+	const navMenuRef = useRef();
+	const [ hasScrollX, setHasScrollX ] = useState( false );
+	const clientWidth = navMenuRef.current ? navMenuRef.current.clientWidth : 0;
+	const scrollWidth = navMenuRef.current ? navMenuRef.current.scrollWidth : 0;
+
+	useLayoutEffect( () => {
+		if ( scrollWidth > clientWidth ) {
+			setHasScrollX( true );
+		} else {
+			setHasScrollX( false );
```

Comment:
minor suggestion
what do you think about 
```es6
useLayoutEffect( () => setHasScrollX( scrollWidth > clientWidth ), [ clientWidth, scrollWidth ] );
```

Feel free to ignore it 👍

Reply:
Yes that is valid.